Source/Classes/PINRemoteImageManager.h

```diff
@@ -542,12 +542,26 @@ typedef void(^PINRemoteImageManagerProgressDownload)(int64_t completedBytes, int
  Cancel a download. Canceling will only cancel the download if all other downloads are also canceled with their associated UUIDs. 
  Canceling *does not* guarantee that your completion will not be called. You can use the UUID provided on the result object to verify
  the completion you want called is being called.
+ @param UUID NSUUID of the task to cancel.
  @param storeResumeData if YES and the server indicates it supports resuming downloads, downloaded data will be stored in the memory
  cache and used to resume the download if the same URL is attempted to be downloaded in the future.
  */
 - (void)cancelTaskWithUUID:(nonnull NSUUID *)UUID storeResumeData:(BOOL)storeResumeData;
 
 /**
+ Cancel a download if the estimated time to complete the download exceeds the passed in value.
+ Canceling will only cancel the download if all other downloads are also canceled with their associated UUIDs.
+ Canceling *does not* guarantee that your completion will not be called. You can use the UUID provided on the result object to verify
+ the completion you want called is being called.
+ @param UUID NSUUID of the task to cancel.
+ @param timeToExceed only cancel the download if the expected time to complete it is greater than timeToExceed
+ @param storeResumeData if YES and the server indicates it supports resuming downloads, downloaded data will be stored in the memory
+ cache and used to resume the download if the same URL is attempted to be downloaded in the future.
+ */
+
+- (void)cancelTaskWithUUID:(nonnull NSUUID *)UUID ifExpectedToExceedTimeInterval:(NSTimeInterval)timeToExceed storeResumeData:(BOOL)storeResumeData;
```

Comment:
I see your point. On one hand, we're "asking" PINRemoteImage to save us some network bandwidth based on best effort measurements. On the other, we believe the caller has a specific plan of action, and part of that plan involves dropping in-flight network requests.

Relatedly, would it be helpful to think of this as a "suspend" operation rather than a "cancel" operation?